### PR TITLE
Add open_dataset to api generation

### PR DESCRIPTION
**Pull request by anissa111**
Addresses #122 

Summary of changes:
- minor formatting changes in api index file (alphabetization, making underline same length as title)
- Adds section for dataset functions in user api
- adds `open_dataset()` to a toctree in user api
- minor formatting changes to `open_dataset()` docstring for better sphinx generation

### CODE REVIEW

File: uxarray/dataset.py [hunk 1]
```diff
@@ -31,12 +31,12 @@ def open_dataset(grid_file, *args, **kw):
...
     --------
-
     Open grid file only
-    >>> mesh = ux.open_dataset("grid_filename.g")
 
+    >>> mesh = ux.open_dataset("grid_filename.g")
 
     Open grid file along with data
+
     >>> mesh_and_data = ux.open_dataset("grid_filename.g", "grid_filename_vortex.nc")
```
Comment (erogluorhan) [inline uxarray/dataset.py:40]: What do you think about adding two other examples, something like the following (to reflect the features Philip demonstraets in his example notebook, i.e. opening 1 grid file and multiple data files, as well as 1 grid file and an arary that holds multiple grid files):

```
Open grid file along with multiple data files (two or more)
>>> mesh_and_data_2 = ux.open_dataset("grid_filename.g", "grid_filename_vortex_1.nc", "grid_filename_vortex_2.nc")

Open grid file along with a list of data files
>>> data_files = ["grid_filename_vortex_1.nc", "grid_filename_vortex_2.nc", "grid_filename_vortex_3.nc"]
>>> mesh_and_data = ux.open_dataset("grid_filename.g", *data_files)
```

**Review by erogluorhan — APPROVED** [review]
Thanks for this! Approved, but please see an inline comment of mine and address it however you see fit.

File: uxarray/dataset.py (hunk 1 above)
Comment (philipc2) [reply uxarray/dataset.py:40]: This seems like a beneficial addition since it may not be immediately apartment on how to load in multiple data files

Comment (anissa111) [reply uxarray/dataset.py:40]: Sounds good! I'll add your suggestions

**Review by philipc2 — APPROVED** [review]
Looks good! Great work!

**Review by michaelavs — APPROVED** [review]

**Pull request merged**